witchcraft/internal/tcpjson/tcp_logger.go

```diff
@@ -106,7 +106,7 @@ func (d *TCPWriter) Write(p []byte) (n int, err error) {
 		n, err := conn.Write(envelope[total:])
 		total += n
 		if err != nil {
-			if nerr, ok := err.(net.Error); !(ok && (nerr.Temporary() || nerr.Timeout())) {
+			if nerr, ok := err.(net.Error); !ok || !nerr.Temporary() {
```

Comment:
I am slightly nervous that a timeout error could also be considered temporary, but looking at the [tls.Conn](https://go.googlesource.com/go/+/go1.16/src/crypto/tls/conn.go#178) code it doesn't seem like that is the case. Mind double checking this logic?

Reply:
I just went ahead and added the explicit check for a timeout as well to be safe.